# Q.1
### A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

- Ho = Diameter of cutlets is same, Da = Db.
- Ha = Diameter of cutlets has significant difference, Da != Db.
- alpha = significance level = 5% = 0.05.
- For above example we use two-sample two-tail method


In [1]:
import pandas as pd
import numpy as np
from scipy import stats
import scipy.stats as st

In [3]:
df = pd.read_csv('Cutlets.csv')
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
Da = pd.Series(data=df['Unit A'])
Da

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
11    7.5169
12    6.9246
13    6.9256
14    6.5797
15    6.8394
16    6.5970
17    7.2705
18    7.2828
19    7.3495
20    6.9438
21    7.1560
22    6.5341
23    7.2854
24    6.9952
25    6.8568
26    7.2163
27    6.6801
28    6.9431
29    7.0852
30    6.7794
31    7.2783
32    7.1561
33    7.3943
34    6.9405
Name: Unit A, dtype: float64

In [5]:
Db = pd.Series(data=df['Unit B'])
Db

0     6.7703
1     7.5093
2     6.7300
3     6.7878
4     7.1522
5     6.8110
6     7.2212
7     6.6606
8     7.2402
9     7.0503
10    6.8810
11    7.4059
12    6.7652
13    6.0380
14    7.1581
15    7.0240
16    6.6672
17    7.4314
18    7.3070
19    6.7478
20    6.8889
21    7.4220
22    6.5217
23    7.1688
24    6.7594
25    6.9399
26    7.0133
27    6.9182
28    6.3346
29    7.5459
30    7.0992
31    7.1180
32    6.6965
33    6.5780
34    7.3875
Name: Unit B, dtype: float64

In [6]:
t_stats,p_value = stats.ttest_ind(Da,Db)

In [7]:
p_value

0.4722394724599501

In [9]:
# For two tail
alpha = 0.05
alpha/2

0.025

- Inference: As (p_value=0.4722) > (α = 0.05); Accept Null Hypothesis i.e. μ1 = μ2. Thus, there is no difference in diameters of cutlets between two units

# Q.2
### A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

- Ho = Average Turn Around Time (TAT) of reports of the laboratories are same.
- Ha = Atleast one average Turn Around Time (TAT) is different.
- alpha = significance level = 5% = 0.05.
- For above example we use Anova test.

In [4]:
df = pd.read_csv('LabTAT.csv')
df

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [5]:
df.isna().sum()

Laboratory 1    0
Laboratory 2    0
Laboratory 3    0
Laboratory 4    0
dtype: int64

In [6]:
f_stat,p_value = st.f_oneway(df['Laboratory 1'],df['Laboratory 2'],df['Laboratory 3'],df['Laboratory 4'])
f_stat,p_value

(118.70421654401437, 2.1156708949992414e-57)

In [7]:
p_value

2.1156708949992414e-57

In [8]:
alpha = 0.05

- Inference: As (p-value=0) < (α = 0.05); Reject Null Hypothesis i.e. Atleast one sample TAT population mean is different Thus there is variance or difference in average Turn Around Time (TAT) of reports of the laboratories on their preferred list.

# Q3.
### Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

- Ho = Male and Female columns are independent.
- Ha = Male and Female columns are dependent.
- alpha = significance level = 5% = 0.05.
- For above example we use Chi-Square test.

In [9]:
df = pd.read_csv('BuyerRatio.csv',index_col = 0)
df

East  West  North  South
Observed Values                          
Males              50   142    131     70
Females           435  1523   1356    750

In [12]:
obs = [[50,142,131,70],
        [435,1523,1356,750]]

In [13]:
test_stats, p, dof, exp = st.chi2_contingency(obs)
test_stats, p, dof, exp

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [15]:
observed = np.array([50, 142, 131, 70, 435, 1523, 1356, 750])
expected = np.array([42.76531299, 146.81287862, 131.11756787, 72.30424052, 442.23468701, 1518.18712138, 1355.88243213, 747.69575948])

In [16]:
test_stats, p_value = st.chisquare(observed,expected,ddof=3)
test_stats, p_value

(1.5959455390914483, 0.8095206646905712)

In [17]:
p_value

0.8095206646905712

In [18]:
alpha = 0.05

- Inference: As (p-value = 0.6603) > (α = 0.05); Accept the Null Hypothesis i.e. Independence of categorical variables Thus, male-female buyer rations are similar across regions and are not related

# Q4.
### TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences.

- Ho = The defective % is independent of centre.
- Ha = The defective % is dependent of centre.
- alpha = significance level = 5% = 0.05.
- For above example we use Chi-square test.

In [20]:
df = pd.read_csv('Costomer+OrderForm.csv')
df

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [21]:
pd.DataFrame([df['Phillippines'].value_counts(),df['Indonesia'].value_counts(),df['Malta'].value_counts(),df['India'].value_counts()])

Error Free  Defective
Phillippines         271         29
Indonesia            267         33
Malta                269         31
India                280         20

In [22]:
obs = [[271,267,269,280],[29,33,31,20]]
obs

[[271, 267, 269, 280], [29, 33, 31, 20]]

In [23]:
test_stat, p, dof, exp = st.chi2_contingency(obs)
test_stat, p, dof, exp

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [24]:
observed = np.array([271,267,269,280,29,33,31,20])
expected = np.array([271.75, 271.75, 271.75, 271.75, 28.25,  28.25,  28.25,  28.25])

In [25]:
test_stat, p_value = st.chisquare(observed, expected, ddof=3)
test_stat, p_value

(3.858960685820355, 0.4254298144535761)

In [26]:
p_value

0.4254298144535761

In [29]:
alpha = 0.05

- Inference: As (p_value = 0.2771) > (α = 0.05); Accept Null Hypthesis i.e. Independence of categorical variables Thus, customer order forms defective % does not varies by centre